### PR TITLE
Assert checks for NaN/infinite values in Skia canvas Dart wrappers

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde @Hixie

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -12,6 +12,21 @@ part of dart.ui;
 // or invalid value to be used by the engine even in release mode, since that
 // would cause a crash. It is, however, acceptable for error messages to be much
 // less useful or correct in release mode than in debug mode.
+//
+// Painting APIs will also warn about arguments representing NaN or infinite
+// coordinates, which can not be rendered by Skia.
+
+bool rectIsValid(Rect rect) {
```
Comment (Hixie) [inline lib/ui/painting.dart]: These should probably be private (prefixed with `_` so they're only visible in this file).

**Comment by Hixie** [comment]
LGTM

File: lib/ui/painting.dart (hunk 1 above)
Comment (jason-simmons) [inline lib/ui/painting.dart]: done

**Pull request merged**